go/libraries/doltcore/sqle/dtables/diff_table.go

```diff
@@ -292,6 +292,23 @@ func (dt *DiffTable) fromCommitLookupPartitions(ctx *sql.Context, hashes []hash.
 	cmHashToTblInfo := make(map[hash.Hash]TblInfoAtCommit)
 	var pCommits []*doltdb.Commit
 	for i, hs := range hashes {
+		headHash, err := dt.head.HashOf()
+		if err != nil {
+			return nil, err
+		}
+		if hs.Equal(headHash) {
+			// from_commit = 'HEAD' needs to match with 'WORKING'
```

Comment:
I had to think about this a bit... so if `from_commit` specifies the HEAD commit, then we also need to include the working root in the parent commits, since that's one step away from the `from_commit`. At first I had read it as "if `HEAD` is specified, use `WORKING` instead.

Might be worth adding a few more words to this comment to make that easier for future readers to understand what's going on here.